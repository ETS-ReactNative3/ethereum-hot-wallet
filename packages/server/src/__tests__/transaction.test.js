// @flow
import micro from 'micro';
import listen from 'test-listen';
import request from 'request-promise';
import MOCK_TRANSACTION from '../__mock-data__/transaction';
import index from '../index';
import { type Handler, type Req, type Res } from '../utils/type.flow';

jest.mock('../utils/web3-helper', () => ({
  extend: () => {},
  middleware: () => (handler: Handler) => (
    req: Req,
    res: Res,
    ...restArgs: any
  ) => {
    req.context = {
      ...req.context,
      web3: {
        extend: () => {},
        eth: {
          getTransaction: () => Promise.resolve(MOCK_TRANSACTION),
          getBlock: () => Promise.resolve({}),
        },
        admin: {
          nodeInfo: () => Promise.resolve({}),
        },
      },
    };
    return handler(req, res, ...restArgs);
  },
}));

it('should return correct json data', async () => {
  const service = micro(index);
  let error;
  try {
    const url = await listen(service);
    const response = await request({
      uri: `${url}/transaction/123`,
    });
    expect(response).toMatchSnapshot();
  } catch (err) {
    error = err;
  }

  expect(error).toBeUndefined();
  service.close();
});
