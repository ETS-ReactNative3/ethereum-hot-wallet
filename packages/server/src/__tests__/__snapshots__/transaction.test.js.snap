// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correct json data 1`] = `"{\\"blockHash\\":\\"0x8e38b4dbf6b11fcc3b9dee84fb7986e29ca0a02cecd8977c161ff7333329681e\\",\\"blockNumber\\":1000000,\\"from\\":\\"0x32Be343B94f860124dC4fEe278FDCBD38C102D88\\",\\"gas\\":50000,\\"gasPrice\\":\\"60000000000\\",\\"hash\\":\\"0xe9e91f1ee4b56c0df2e9f06c2b8c27c6076195a88a7b8537ba8313d80e6f124e\\",\\"nonce\\":17387,\\"to\\":\\"0xDf190dC7190dfba737d7777a163445b7Fff16133\\",\\"value\\":\\"437194980000000000\\"}"`;
